src/workerd/io/observer.h

```diff
@@ -160,6 +163,37 @@ class RequestObserver: public kj::Refcounted {
   }
 };
 
+// At the conclusion of an invocation we need to dispatch the outcome
+// event to the streaming tail worker (if any). We essentially have a race
+// between the completion of waitUntil tasks and the completion of the
+// deferred proxy (if any). The request is done whenever both of these are
+// finished. However, those end up being two separate (and unrelated) branches
+// of the promise tree set up by an invocation.
```

Comment:
FWIW, this comment is mostly relevant to just the fetch handler. Other handlers *might* have similar characteristics and all handlers do need to wait for `waitUntil` tasks but the exact placement of the outcome event varies from handler to handler. It might make sense to move this comment back into the fetch handler where this is used.